### Q-6. A pharmaceutical company is developing a drug that is supposed to reduce blood pressure. They conduct a clinical trial with 100 patients and record their blood pressure before and after taking the drug. The company wants to know if the change in blood pressure follows a normal distribution.
 
 https://drive.google.com/file/d/1mCjtYHiX--mMUjicuaP2gH3k-SnFxt8Y/view?usp=share_


In [8]:
import numpy as np
from scipy.stats import shapiro,power_divergence
import statsmodels.stats.power as smp


# Load the data
data = np.loadtxt("data.csv", delimiter=",", skiprows=1)

# Perform the Shapiro-Wilk test
stat, pvalue = shapiro(data)

# Print the results
print("Test statistic:", stat)
print("p-value:", pvalue)

# Conduct a power analysis
effect_size = 0.5
alpha = 0.05  # Significance level is 0.05
n = len(data)

# Calculate the required sample size for a given power
required_sample_size = smp.tt_solve_power(effect_size, nobs=None, alpha=alpha, power=power, alternative='two-sided')

# Print the required sample size
print("Required Sample Size:", required_sample_size)

# Calculate the power
power = power_divergence(data, 'shapiro', effect_size=effect_size, nobs=n, alpha=alpha)[0]

# Print the power
print("Power:", power)

Test statistic: 0.792026162147522
p-value: 2.812811387868261e-19
Required Sample Size: 53.94061742677978


TypeError: power_divergence() got an unexpected keyword argument 'effect_size'

In [10]:
import numpy as np
from scipy.stats import shapiro

# Load the data
data = np.loadtxt("data.csv", delimiter=",", skiprows=1)
# Perform the Shapiro-Wilk test
stat, pvalue = shapiro(data)

# Print the results
print("Test statistic:", stat)
print("p-value:", pvalue)

# Conduct a power analysis
effect_size = 0.5
alpha = 0.05
n_simulations = 1000
significant_count = 0

# Simulate datasets and perform the Shapiro-Wilk test
for _ in range(n_simulations):
    simulated_data = np.random.normal(loc=0, scale=1, size=len(data))  # Example: normal distribution
    sim_stat, sim_pvalue = shapiro(simulated_data)
    if sim_pvalue < alpha:
        significant_count += 1

# Calculate power as the proportion of significant results
power = significant_count / n_simulations

# Print the power
print("Power:", power)


Test statistic: 0.792026162147522
p-value: 2.812811387868261e-19
Power: 0.055


We can make the following conclusions:

Test statistic: The test statistic for the Shapiro-Wilk test is 0.792026162147522. This value represents the calculated test statistic based on the data.

p-value: The p-value for the Shapiro-Wilk test is 2.812811387868261e-19. The p-value is a measure of the evidence against the null hypothesis. In this case, the extremely small p-value suggests strong evidence to reject the null hypothesis.

Power: The power of the test is 0.055. The power represents the probability of correctly rejecting the null hypothesis when it is false. In this case, the low power value suggests that the test has a low ability to detect departures from the null hypothesis, assuming that such departures exist.

Based on these results, we can conclude the following:

1. The change in blood pressure after taking the drug does not follow a normal distribution. This conclusion is supported by the very small p-value, which indicates strong evidence against the null hypothesis of normality.

2. The Shapiro-Wilk test has a low power of 0.055, indicating that it has a low ability to detect departures from normality, assuming they exist. This means that even if the change in blood pressure does not follow a normal distribution, the test may not be able to identify it reliably.

In summary, the results suggest that the change in blood pressure after taking the drug does not follow a normal distribution, and the Shapiro-Wilk test used in this analysis has a low power to detect departures from normality.




